reentrancy_guard_example/deposit_example/src/deposit_actor.rs

```diff
@@ -0,0 +1,264 @@
+            })?;
+
+            Ok(())
+        })
+    }
+
+    fn check_balance(rt: &impl Runtime, address: Address) -> Result<TokenAmount, ActorError> {
+        rt.validate_immediate_caller_accept_any()?;
+
+        let address = match resolve_address(&address) {
+            Some(address) => address,
+            None => {
+                return Err(actor_error!(
+                    assertion_failed,
+                    "Err: cannot resolve address"
+                ))
+            }
+        };
+
+        let st: State = rt.state()?;
+
+        let map = map::make_map_with_root::<_, TokenAmount>(&st.deposit_funds, rt.store())
+            .map_err(|e| {
+                e.downcast_default(
+                    ExitCode::USR_ASSERTION_FAILED,
+                    "Failed to create map from Cid.",
+                )
+            })?;
+
+        let key: BytesKey = address.to_bytes().into();
+
+        match map.get(&key).unwrap() {
+            Some(val) => Ok(val.to_owned()),
+            None => Ok(TokenAmount::zero()),
+        }
+    }
+
+    fn withdraw(rt: &impl Runtime) -> Result<(), ActorError> {
+        rt.validate_immediate_caller_accept_any()?;
+
+        let sender = Actor::msg_sender();
+        let st: State = rt.state()?;
+
+        let mut map = map::make_map_with_root::<_, TokenAmount>(&st.deposit_funds, rt.store())
+            .map_err(|e| {
+                e.downcast_default(
+                    ExitCode::USR_ASSERTION_FAILED,
+                    "Failed to create map from Cid.",
+                )
+            })?;
+
+        let key: BytesKey = sender.to_bytes().into();
+
+        let balance = match map.get(&key).unwrap() {
+            Some(val) => val.to_owned(),
+            None => TokenAmount::zero(),
+        };
+
+        if balance.is_zero() {
+            return Err(actor_error!(
+                assertion_failed,
+                "No available balance to be withdrawn"
+            ));
+        }
+
+        // Method num = FIRST_EXPORTED_METHOD_NUMBER(16777216) + 1 = 16777217
+        rt.send_simple(&sender, 16777217, None, balance)?;
+
+        rt.transaction(|st: &mut State, _| {
+            map.set(key, TokenAmount::zero()).map_err(|e| {
+                e.downcast_default(
+                    ExitCode::USR_ASSERTION_FAILED,
+                    "Failed to reset senders balance.",
+                )
+            })?;
+
+            st.deposit_funds = map.flush().map_err(|e| {
+                e.downcast_default(ExitCode::USR_ASSERTION_FAILED, "Failed to flush map.")
+            })?;
+
+            Ok(())
+        })?;
+
+        Ok(())
+    }
+
+    fn withdraw_guarded(rt: &impl Runtime) -> Result<(), ActorError> {
+        rt.validate_immediate_caller_accept_any()?;
+
+        let sender = Actor::msg_sender();
+        let st: State = rt.state()?;
+
+        rt.transaction(|st: &mut State, _| {
+            st.reentrancy_guard.non_reentrant_before();
+            Ok(())
+        })?;
```

Comment:
First check non_reentrant_before, then extract sender from the message.

```suggestion
        let st: State = rt.state()?;

        rt.transaction(|st: &mut State, _| {
            st.reentrancy_guard.non_reentrant_before();
            Ok(())
        })?;

        let sender = Actor::msg_sender();
```

Reply:
Changed [here](https://github.com/Ethernal-Tech/filecoin-actor-library/blob/ReentrancyGuard_actor_and_tests/reentrancy_guard_example/deposit_example/src/deposit_actor.rs#L195).